Ore distribution: Deeper iron, diamond and mese block, tune gold

paramat opened the pull request:
Affects non-mgv6 mapgens only.

Relevant to #1168 and #1681
Makes progression less ridiculously easy while also encouraging digging deeper.
Compensates for the 3D noise tunnels which make vertical travel easier.

Iron at y = 0 was far too easy to find.
Adjust gold lower region to be twice the depth of highest level, like
all other ores.
Diamond at y = -500 was too easy to progress to.
Make diamond and mese block deeper to create a depth progression from
mese crystal to diamond, to reflect tool progression.
This all creates a satisfying 2^n depth progression, with lower regions
being at twice the depth of highest levels.

The new depths are:

Ore. Highest. Density increase
Coal 64 -128
Iron -64 -128
Tin -128 -256
Copper -128 -256
Gold -256 -512
Mese crystal -512 -1024
Diamond -1024 -2048
Mese block -2048 -4096

Apart from the special case of coal, the depths progress as 2^n numbers doubling per value level of ore:
Iron -64
Tin, Copper (for bronze) -128
Gold -256
Mese crystal -512
Diamond -1024
Mese block -2048

The y of density increase is now double these depths for each.

Another advantage to not having iron at y = 0 is that the depths of iron->steel and tin/copper->bronze could be reversed in future to represent steel tools being more valuable than bronze tools, as they should be (but are not). This would have to wait for steel tools to be made more capable than bronze tools, as they should be (but are not).

Ezhh commented:
This is worth merging as it since it deals with a current issue, and I'd consider it a separate issue to the larger tool re-balance one.

paramat commented:
Yes ok.

SmallJoker commented:
Did I miss anything? With this new change, the gold ore densities from -256 to -1023 and -512 to bottom will overlap from -512 to -1023, which means they will generate there much more often.

paramat commented:
Oops.

paramat commented:
Fixed.

paramat commented:
WIP, thinking about deeper diamond and mese block now.

paramat commented:
Updated, rewrote first post. probably needs re-approval.
Almost everyting is now at twice the depth of 0.4.16 levels, and the 2^n numbers are lovely.

paramat commented:
@Ezhh this has changed so you may want to re-assess your approval.

Ezhh commented:
I'm fine with depth increases.
